web-server/conf/routes

```diff
@@ -3,19 +3,19 @@ GET        /                                 codesearch.web.controllers.Applicat
 GET        /*path/                           codesearch.web.controllers.Application.untrail(path: String)
 
 GET        /haskell                          codesearch.web.controllers.HackageSearcher.index
-GET        /haskell/search                   codesearch.web.controllers.HackageSearcher.search(query: String ?= "", filter: String ?= "", insensitive: String ?= "", space: String ?= "", precise: String ?= "", sources: String ?= "on", page: String ?= "1")
+GET        /haskell/search                   codesearch.web.controllers.HackageSearcher.search(query: String ?= "", filter: Option[String], insensitive: String ?= "", space: String ?= "", precise: String ?= "", sources: String ?= "on", page: String ?= "1")
```

Comment:
Why not make all the parameters optional?

Reply:
What for? Do we have all the fields, either true or false, and the query doesn't exactly have to be optional?